### ISSUE TITLE
Prohibit !voice host ing someone without Yes Theory role

### ISSUE THREAD

**Issue by geisterfurz007** [Yes-Theory-Fam/yesbot-ts#145]
The voice command can only be used by users with the Yes Theory role. Currently `!voice host` allows an room owner to give their access to someone without that role, rendering the room frozen because the new owner cannot interact with the room.

The bot uses this code to check if the new host has the Yes Theory role:

https://github.com/Yes-Theory-Fam/yesbot-ts/blob/0d6cd14e87aec5f493eca7976c9503e1197e2cc4/src/programs/VoiceOnDemand.ts#L336-L342

It contains a subtle but bad mistake: `member` should be the mentioned user.

## Implementation

Change
https://github.com/Yes-Theory-Fam/yesbot-ts/blob/0d6cd14e87aec5f493eca7976c9503e1197e2cc4/src/programs/VoiceOnDemand.ts#L313
```diff
- const mentionedUser = message.mentions.users.first();
+ const mentionedMember = message.mentions.members.first();
```
and rename the variable in other places as well (there actually is a difference between a Member (which contains server related information like roles) and a user (which is a "plain" user of Discord, not specific to a certain server)).

In the condition mentioned above, replace member with `mentionedMember` to check the roles of the correct person.

Lastly, you need to fix a call further down:

https://github.com/Yes-Theory-Fam/yesbot-ts/blob/0d6cd14e87aec5f493eca7976c9503e1197e2cc4/src/programs/VoiceOnDemand.ts#L346

As the function expects a User object, not a GuildMember, you need the following change:

```diff
-  await transferOwnership(repo, mapping, mentionedUser, memberVoiceChannel);
+ await transferOwnership(repo, mapping, mentionedMember.user, memberVoiceChannel);
```

**Comment by geisterfurz007** [comment]
This was fixed in https://github.com/Yes-Theory-Fam/yesbot-ts/commit/ea7fa734c1a30b33c3be17fd05dcfe8ce1f46d12

**Comment by geisterfurz007** [comment]
This was not fixed in https://github.com/Yes-Theory-Fam/yesbot-ts/commit/ea7fa734c1a30b33c3be17fd05dcfe8ce1f46d12!

**Comment by geisterfurz007** [comment]
Closed in fdb687cf3094ef90ffb6a54b035e68201893ef74
